cmd/antrea-agent/agent.go

```diff
@@ -151,6 +152,13 @@ func run(o *Options) error {
 
 	go networkPolicyController.Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go metrics.InitializePrometheusMetrics(
```

Comment:
Will api server in antrea agent need a new handler for /metrics handle to work? 
https://github.com/vmware-tanzu/antrea/blob/775ce88e3d6b372c39df0e5fd95405a7381e5459/pkg/agent/apiserver/apiserver.go#L54

Reply:
Please ignore this. Found the install handle code for /metrics at kubernetes/staging/src/k8s.io/apiserver/pkg/server

Reply:
A follow up: Don't we need to register antrea specific metrics like the default metrics that are already existing under /metrics handler? 
https://github.com/kubernetes/kubernetes/blob/26d497ac57b4fdcfb564051d45757feaa75ec66e/staging/src/k8s.io/apiserver/pkg/server/routes/metrics.go

Reply:
>>As for /metrics: each component/package which exposes Prometheus metrics via >>Prometheus client lib will be exposed within the same /metrics URL. I specifically allowed >>disabling GO, process metrics to reduce irrelevant clutter (and also because Calico does so ;)
Thanks for the response. Going after apimetrics in the above file, found that there is legacyRegistry package. k8s.io/component-base/metrics/legacyregistry/registry.go
Above is a wrapper over prometheus registry objects. That seems to be there for metrics stability across versions of k8s/antrea as they can be modified in future (name/type/labels). That is my understanding. https://github.com/kubernetes/enhancements/pull/946/files/1ecb5330ef1254543ac7b462b66f04645f08fc80
Can that be used instead of direct prometheus objects? Let me know your thoughts. 
@antoninbas

Reply:
I think that using the k8s wrapper is better. I would add a separate issue for using it, and have Prometheus integration in the meantime via the registry objects.